src/http_async.rs

```diff
@@ -49,16 +55,12 @@ pub async fn download_file_progress(
     let mut stream = res.bytes_stream();
 
     while let Some(item) = stream.next().await {
-        let chunk = match item {
-            Ok(v) => v,
-            Err(e) => return Err(format!("Error while downloading file: {e}")),
-        };
-        if let Err(e) = file.write_all(&chunk) {
-            Err(format!("Error while writing to file: {e}"))?
-        }
+        let chunk = item.map_err(|e| format!("Error while downloading file: {e}"))?;
+        file.write_all(&chunk)
+            .map_err(|e| format!("Error while writing to file: {e}"))?;
         let new = min(downloaded + (chunk.len() as u64), total_size);
         downloaded = new;
-        pb.set_position(new);
+        pb.set_position(new); // This should be done in a separate thread if it's a spinner
```

Comment:
this should be an issue not a comment in code
aside from that it's fine in the same thread, the spinner won't be perfectly smooth but in the end its just an indicator that stuff is happening.